### PR TITLE
[RLlib] Don't test SimpleServing with connectors

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
Signed-off-by: Artur Niederfahrenhorst <artur@anyscale.com>

## Why are these changes needed?

Our current connector sample collection logic does not work with external envs.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by gjoliver — APPROVED** [review]

File: rllib/tests/test_nested_observation_spaces.py [hunk 1]
```diff
@@ -458,6 +458,10 @@ def test_nested_dict_vector(self):
         )
 
     def test_nested_dict_serving(self):
+        # TODO: (Artur) Enable this test again for connectors if discrepancies
+        #  between EnvRunnerV2 and ExternalEnv are resolved
+        if PGConfig().enable_connectors is True:
```
Comment (kouroshHakha) [inline rllib/tests/test_nested_observation_spaces.py:463]: nit: isn't enable_connectors always a boolean? so it should be just `if PGConfig().enable_connectors:`

File: rllib/tests/test_nested_observation_spaces.py [hunk 2]
```diff
@@ -472,6 +476,10 @@ def test_nested_tuple_vector(self):
         )
 
     def test_nested_tuple_serving(self):
+        # TODO: (Artur) Enable this test again for connectors if discrepancies
+        #  between EnvRunnerV2 and ExternalEnv are resolved
+        if PGConfig().enable_connectors is True:
```
Comment (kouroshHakha) [inline rllib/tests/test_nested_observation_spaces.py:481]: same thing here?

**Review by kouroshHakha — APPROVED** [review]
One nit. Thanks man

File: rllib/tests/test_nested_observation_spaces.py [hunk 3]
```diff
@@ -472,6 +476,10 @@ def test_nested_tuple_vector(self):
         )
 
     def test_nested_tuple_serving(self):
+        # TODO: (Artur) Enable this test again for connectors if discrepancies
+        #  between EnvRunnerV2 and ExternalEnv are resolved
```
Comment (kouroshHakha) [inline rllib/tests/test_nested_observation_spaces.py:480]: actually a cleaner code would be:

```
if not PGConfig().enable_connectors:
     self.do_test_nested_dict(lambda _: SimpleServing(NestedDictEnv()))
```

**Review by avnishn — APPROVED** [review]

**Pull request merged**